geps/gep-1897.md

```diff
@@ -125,7 +132,251 @@ This GEP is the outcome of the TLS use cases #4 and #5 in
 
 ## API
 
-Details deferred until we reach consensus on what we want to do, and why we want to do this.
+To allow the gateway client to know how to connect to the backend pod, when the backend pod has its own
+certificate, we implement a metaresource that is already mentioned as an example in
+[GEP-713: Metaresources and PolicyAttachment](https://gateway-api.sigs.k8s.io/geps/gep-713/), and as a hypothetical in the
+[Policy Attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/#direct-policy-attachment)
+documentation.
+
+This metaresource is named TLSConnectionPolicy.  In this document, because naming is hard, we chose to retain the
+name TLSConnectionPolicy to advertise alignment with a previously discussed naming choice, but a new name may be
+substituted without blocking acceptance of the content of the API change.
+
+The selection of the applicable Gateway API persona is important in the design of this proposal, because each
+persona is assigned a role which handles specific Gateway API resources.  TLSConnectionPolicy is used by the application
+developer Gateway API persona to convey client certificate settings used in the TLS handshake from Gateway to backend,
+because this persona handles resources involved with application access and configuration, such as Routes and Services.
```

Comment:
I think the important part here is to consider whose _intent_ we are representing in this API. I suggest something like this:
```suggestion
persona is assigned a role which handles specific Gateway API resources.  TLSConnectionPolicy is used by the application
developer Gateway API persona to signal what the application developer _expects_ connections to the application to look like from a TLS perspective. 
Only the application developer can know what the application expects, so this it's important that this configuration be owned by that persona.
```
The important part here is that the _application running inside the Pods_ expects connections to look a certain way - the TLSConnectionPolicy object is about having a way for the application developer to _explicitly_ describe that expectation.

We can use this way of looking at things to make calls about what should be included in the TLSConnectionPolicy.

* The _backend_ should already be managing its own cert somehow, so we shouldn't need to represent that here.
* The Gateway, as a TLS client, needs to know some things about how to connect to the service (most of these are handled automatically for _end user clients_ by the browser):
  * that TLS should be attempted at all (a browser will use the scheme to determine this, but Gateways don't have that information available _at an API level_ - it could be intuited from some config, but it should be explicitly configured.)
  * What CA should be used (a browser usually will use the system bundled CAs, and allow adding extras, it's _much_ more common to use either individual self-signed certificates or an internal CA for Gateway-to-internal-backend use cases)
  * What version of TLS should be expected (this is not strictly necessary, because TLS is a negotiated protocol, but it is _very_ nice to have because it will allow Gateway implementations to do things like disallow TLS 1.1, force only TLS 1.3, and so on).

While the Gateway as a TLS client does need to know if it should pass a client certificate, that should be configured on the Gateway, by the Gateway owner, so that the client and server have their chain of custody properly managed. (see https://github.com/kubernetes-sigs/gateway-api/pull/2113/files#r1268891063 for some more thoughts on this from me).

Reply:
@robscott I have added to the API to describe an idea that if the backend requires TLS and the Gateway hasn't been configured for it, then it should be possible to use system trusted certs.  I think this addresses your concern.  I don't feel it is a problem for 1 and 2 to be done by the same person.

@youngnick I agree with what you've written here and can update the API.

Reply:
The first two are already there:
That TLS should be attempted at all -> the TLSConfigPolicy is there for the targetRef.
What CA to use -> TrustedCACertRefs

I will add the TLS versions.